### PR TITLE
Fix for failing Node shutdown/reboot commands

### LINKED ISSUE #69: shutdown/reboot node not working
I noticed the "shutdown node" and "reboot node" command are not working. 
By pressing either of the buttons under _Objects>Proxmox.0>node_myproxmox_ I get an error like this in the log

`(1101666) could not Find name in VMs: {"data":[{"disk":0,"maxcpu":4,"mem":0,"type":"qemu","cpu":0,"id":"qemu/200","vmid":200,"status":"stopped","uptime":0,"maxmem":6442450944,"name":"Dockerbuntu","nod...`

By digging in the main.js I see in line 65
`let vm = vms.find(vm => vm.name === vmname);`
that throws this error a little later

I don't understand the whole code but I  try to guess what happens: it seems as if iobroker just delivers the nodename(here _myproxmox_), but obvioulsy no vm.name - which is true as it is not available in this part of the objects-tree.

EDIT: 
The logs after the curly brace are more or less random, even if press the buttons in short cycles:

```
proxmox.0 | 2020-08-04 22:41:38.486 | error | (1101666)  could not Find name in VMs:  {"data":[{"disk":0,"maxcpu":4,"mem":0,"type":"qemu","cpu":0,"id":"qemu/200","vmid":200,"status":"stopped","uptime":0,"maxmem":6442450944,"name":"Dockerbuntu","nod
-- | -- | -- | --
proxmox.0 | 2020-08-04 22:41:37.308 | error | (1101666)  could not Find name in VMs:  {"data":[{"maxmem":6442450944,"id":"qemu/200","vmid":200,"status":"stopped","uptime":0,"disk":0,"type":"qemu","mem":0,"maxcpu":4,"cpu":0,"netin":0,"maxdisk":0,"ne
proxmox.0 | 2020-08-04 22:40:58.817 | error | (1101666) could not Find name in VMs:  {"data":[{"maxmem":6442450944,"disk":0,"cpu":0,"mem":0,"maxcpu":4,"type":"qemu","vmid":200,"id":"qemu/200","status":"stopped","uptime":0,"maxdisk":0,"diskread":0
proxmox.0 | 2020-08-04 22:40:57.691 | error | (1101666) could not Find name in VMs:  {"data":[{"maxmem":6442450944,"disk":0,"cpu":0,"maxcpu":4,"mem":0,"type":"qemu","vmid":200,"id":"qemu/200","status":"stopped","uptime":0,"maxdisk":0,"diskread":0
proxmox.0 | 2020-08-04 22:40:56.807 | error | (1101666) could not Find name in VMs:  {"data":[{"disk":0,"cpu":0,"type":"qemu","mem":0,"maxcpu":4,"vmid":200,"id":"qemu/200","status":"stopped","uptime":0,"maxmem":6442450944,"name":"Dockerbuntu","di
proxmox.0 | 2020-08-04 22:40:55.879 | error | (1101666) could not Find name in VMs:  {"data":[{"uptime":0,"status":"stopped","id":"qemu/200","vmid":200,"type":"qemu","mem":0,"maxcpu":4,"cpu":0,"disk":0,"maxmem":6442450944,"node":"proxmox","templa
```

### PR DESCRIPTION
**Pull request by ThetaGamma**
1.) fixed wrong command send in proxmox.js (https://pve.proxmox.com/pve-docs/api-viewer/#/nodes/{node}/status)
2.) moved search for VMs to the right place and fixed missing population of variable "node"

fixes #69

tested OK on actual iobroker with proxmox
thanks to @herobone for helping me

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -54,25 +54,33 @@ adapter.on('stateChange', function (id, state) {
...
         let type = vmdata.split('_')[0];
-        let vmname = vmdata.split('_')[1];
-        let command = id.split('.')[3]
-        let vmid, node;
-
-        adapter.log.debug('state changed ' + command + ' : type:  ' + type + ' vmname: ' + vmname);
-        // get vm vid
+        // "if" is not working, could not find out why -> dirty workaround
+        //if (type === 'lxc' || type === 'qemu') {
+            let vmname = vmdata.split('_')[1];
```
Comment (Apollon77) [inline main.js:59]: is this really the same array index for name and node?
Please (even if it was that way before) split twice :-)

Comment (ThetaGamma) [reply main.js:59]: sounds weird, but it is the same index. To verify it: If you enable the logging in line 50 @ (main.js) and 

1. press "start"-button on a proxmox vm called "myfinevm" listed in iobroker, you'll get as value the variable "id":
_'stateChange proxmox.0.qemu_myfinevm.start'_ 
which leads to the vmname => `vmdata.split('_')[1]; = "myfinevm"`
2. press "shutdown" on the node with hostname 'mybigmachine', you'll get as value the variable "id":
_'stateChange proxmox.0.node_mybigmachine.shutdown'_
which leads to the nodename => `vmdata.split('_')[1]; = "mybigmachine"` 

I tried to catch this a bit more elegant with the out-commented "if"-clause, but my javascript-Google-fu was not good enough to get this working ;-)

Does this answer your question? I actually don't know what you mean by "Please split twice" 

kannst mir auch auf deutsch antworten, dann verstehe ich evtl leichter was du meinst ;-)

Comment (ThetaGamma) [reply main.js]: Update: although Google-Fu did not work, asking a real person did help => the actual commit contains the working _if_-clause

**Comment by Apollon77** [comment]
Machst Du dann ein npm update bitte?

**Comment by MeisterTR** [comment]
Jep mach ixh heute abend fertig

**Pull request merged**